test/distributed/test_c10d_nccl.py

```diff
@@ -625,11 +652,11 @@ def allreduce(tensors):
         # Barrier will ensure that all previous work is completed
         pg.barrier().wait()
 
-        for i in range(2, self.num_gpus + 1):
+        for i in range(1, len(local_device_ids) + 1):
             for j in range(i):
                 # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
                 self.assertEqualIgnoreType(
-                    torch.tensor([float(i * (i + 1) / 2)]), tensors_list[i - 2][j]
+                    torch.tensor([(j + 1) * self.world_size]), tensors_list[i - 1][j]
```

Comment:
N00b question, why do we not compare against tensors in works?

Reply:
hmmm sorry what do you mean "tensors in works"?

Reply:
I mean the list `works` define in L648